### ISSUE TITLE
(documentation) Provide example of plaintext usage

### ISSUE THREAD

**Issue by sarnobat** [shundhammer/qdirstat#159]
I'm able to generate the txt file:

```
sudo qdirstat-cache-writer -v "$PWD" /dev/stdout | tee cache.txt
```

but I can't read it in to the tool:

```
qdirstat --cache cache.txt
```

### error output

```
Logging to /tmp/qdirstat-sarnobat/qdirstat.log
libGL error: No matching fbConfigs or visuals found
libGL error: failed to load driver: swrast
zsh: segmentation fault  qdirstat -c cache.txt
```

What am I doing wrong? I'd love to stick to good old plaintext.

**Comment by shundhammer** [comment]
https://github.com/shundhammer/qdirstat/blob/master/scripts/qdirstat-cache-writer#L3-L35

```
# qdirstat-cache-writer - script to write QDirStat cache files from cron jobs
#
# QDirStat can read its information from cache files. This is a lot faster than
# reading all the directories in a directory tree and obtaining detailed
# information (size, type, last modification time) for each file and directory
# with the opendir() / readdir() and lstat() system calls for each individual
# file and directory.
#
# QDirStat can also write those cache files ("Write Cache File..." from the
# "File" menu), but the whole point of cache files is being able to do that in
# the background when the user does not have to wait for it - like in a cron
# job running in the middle of the night. QDirStat itself cannot be used to do
# that because it is a KDE program and thus an X program that needs access to
# an X display - which cron does not provide.
#
# This is what this Perl script is for.
#
# Usage:
#	qdirstat-cache-writer [-lvdh] <directory> [<cache-file-name>]
#
#	If not specified, <cache-file-name> defaults to ".qdirstat.cache.gz"
#	in <directory>.
#
#	If <cache-file-name> ends with ".gz", it will be compressed with gzip.
#	qdirstat can read gzipped and plain text cache files.
#
#	-l	long format - always add full path, even for plain files
#	-m	scan mounted filesystems (cross filesystem boundaries)
#	-v	verbose
#	-d	debug
#	-h	help (usage message)
```

**Comment by shundhammer** [comment]
BTW that whole idea of letting the complete cache file scroll by on stdout is nonsensical; even more so without using a paginator like `less`. Even in the most trivial case, it's way too much text to read; in the best case, you can get a quick glimpse to see if it looks valid in general.

For this kind of operation, use `tail -f` or `tail -F` on the resulting file, not that `| tee` pipeline.

**Comment by sarnobat** [comment]
## Solution

I forgot to remove `-v` which I'd been using when playing around. (side note: maybe verbose output should go to stderr instead)

Works:

```
sudo qdirstat-cache-writer  "$PWD" /dev/stdout | tee cache.txt
qdirstat -c cache.txt
```

**Comment by shundhammer** [comment]
> side note: maybe verbose output should go to stderr instead

No. It's not error output. If you simply don't get creative and use it as documented, there is no problem.

**Comment by sarnobat** [comment]
https://mehulkar.com/blog/2017/11/stdout-vs-stderr/

> I’ve always thought that stderr for a program is to display errors and stdout is for “everything else”. But it turns out stdout is meant to be program output and stderr is meant to be… “everything else”. This was conceptually difficult for me until I thought of an example.

https://unix.stackexchange.com/questions/331611/do-progress-reports-logging-information-belong-on-stderr-or-stdout

> The standard error stream, which is used for error messages and diagnostics issued by the program.

> By the principle of exclusion, it can only go to stderr. Yes, I know you asked about an official specification, which I cannot present you beyond the link to the POSIX specification, given by Stephen Kitt, which states that stderr is for diagnostic purposes.

> So. Nothing except the real "payload" of your program belongs on stdout. If your program has no output, then nothing should go to stdout. This leaves stderr for everything else, including progress reports.

> This doesn't limit its use to error messages only. I would consider progress information as diagnostic output, so it belongs on standard error.

**Comment by shundhammer** [comment]
You still don't get it. The **PAYLOAD** of _qdirstat-cache-writer_ is the cache file, and it goes to that file that you specify on the command line - **UNLESS YOU EXPLICITLY SABOTAGE THAT** by specifying `/dev/stdout`, for the questionable purpose of seeing it scroll by.

I keep telling you: Had you used it like described in the documentation, it would have worked just fine. But that `/dev/stdout` **in combination** with the `-v` flag killed it.

And adding insult to injury, you proved that the entire approach was useless because _you hadn't seen those `-v` messages scrolling by in the masses of text that kept flooding you.

----------------

You are arguing for the sake of arguing. You are proactively wasting my time. I will not cater to snowflakes who cannot bear being called out when they are wrong. I will not introduce compromises to my code for somebody's hurt feelings. I will not optimize for the pathological case.
